packages/oauth/oauth-provider/src/client/client-manager.ts

```diff
@@ -383,11 +373,33 @@ export class ClientManager {
       // > accordingly.
     }
 
+    if (metadata.authorization_details_types?.length) {
+      const dupAuthDetailsType =
+        metadata.authorization_details_types.find(isDuplicate)
+      if (dupAuthDetailsType) {
+        throw new InvalidClientMetadataError(
+          `Duplicate authorization_details_type "${dupAuthDetailsType}"`,
+        )
+      }
```

Comment:
I have no issue with it, but I'm curious what spurred all the new "duplicate" checks?

Reply:
I got the code in for scopes, then realized it was really easy to add for some other checks. Just trying to enforce better quality for clients, while providing good feelback to devs.